File: pi.py
import math
import time

# Start measuring time
start_time = time.time()

# Initialize denominator
k = 1

# Initialize sum
s = 0

iterations = 10**6  # 1 000 000 ciklus

for i in range(iterations):
    if i % 2 == 0:
        s += 4 / k
    else:
        s -= 4 / k
    k += 2

# Print results in normal float format with many decimals
print(f"{s:.15f}")
print(f"{(math.pi - s):.15f}")

# Calculate and print elapsed time
end_time = time.time()
elapsed_time = end_time - start_time
print(f"A kód futási ideje: {elapsed_time:.2f} másodperc")

File: pi.cpp
#include <iostream>
#include <cmath>
#include <iomanip>
#include <chrono>

int main() {
    auto start_time = std::chrono::high_resolution_clock::now();
    long long k = 1;
    double s = 0.0;
    const long long iterations = 10000000000;
    std::cout << std::fixed << std::setprecision(15);
    for (long long i = 0; i < iterations; ++i) {
        if (i % 2 == 0) {
            s += 4.0 / k;
        } else {
            s -= 4.0 / k;
        }
        k += 2;
       }
    std::cout << s << std::endl;
    std::cout << (M_PI - s) << std::endl;
    auto end_time = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> elapsed = end_time - start_time;
    std::cout << "A kód futási ideje: " << std::fixed << std::setprecision(5)
              << elapsed.count() << " másodperc" << std::endl;

    return 0;
}